platform/wayland/cog-view-wl.c

```diff
@@ -214,8 +212,7 @@ cog_wl_view_create_backend(CogView *view)
 bool
 cog_wl_view_does_image_match_win_size(CogWlView *view)
 {
-    CogWlPlatform *platform = (CogWlPlatform *) cog_platform_get();
-    CogWlViewport *viewport = COG_WL_VIEWPORT(platform->viewport);
+    CogWlViewport *viewport = COG_WL_VIEWPORT(cog_view_get_viewport((CogView *) view));
     return view->image && wpe_fdo_egl_exported_image_get_width(view->image) == viewport->window.width &&
            wpe_fdo_egl_exported_image_get_height(view->image) == viewport->window.height;
```

Comment:
We also need to check that the viewport is valid, because `cog_view_get_viewport()` may return `NULL`:

```suggestion
    return viewport && view->image &&
           wpe_fdo_egl_exported_image_get_width(view->image) == viewport->window.width &&
           wpe_fdo_egl_exported_image_get_height(view->image) == viewport->window.height;
```

(Or whichever indentation the style checker prefers.)